### ISSUE TITLE
Vercel builder encounters path error

### ISSUE THREAD

**Issue by djeusette** [jaydenseric/apollo-upload-client#211]
Hi there,

I reach out for help as Vercel's support is quite slow at the moment (I'm still waiting for them to get back to me).  I'm building a Next.js app that relies on your library.  Everything works fine in dev but deploying to staging/prod has become impossible since the addition of apollo-upload-client.

The logs on Vercel's side are:

```
00:01:17.135 | Error: File /vercel/5b518e27/node_modules/apollo-upload-clientpublic/ReactNativeFile does not exist.
00:01:17.135 | at Job.emitDependency (/vercel/42f2bbfe8d152ee9/.build-utils/.builder/node_modules/@vercel/next/dist/index.js:9092:23)
00:01:17.135 | at /vercel/42f2bbfe8d152ee9/.build-utils/.builder/node_modules/@vercel/next/dist/index.js:9121:36
00:01:17.135 | at async Promise.all (index 0)
```

Note the path issue: `/vercel/5b518e27/node_modules/apollo-upload-clientpublic/ReactNativeFile`.  A slash is definitely missing between `apollo-upload-client` and `public`.

I'm not 100% sure where this error comes from but among all the libraries I use no other ever triggered this kind of issue, hence my message.  Would you be so kind as to point me in the right direction?  

Thanks!

**Comment by jeroenvisser101** [comment]
@djeusette I tried to get this resolved and failed for 2 hours, but when I removed ESM support (ditch all .mjs files, remove `exports` key from `extract-files` and `apollo-upload-client`, building on Vercel completes again).

This might be related to https://github.com/vercel/next.js/issues/8113

**Comment by jaydenseric** [comment]
I've seen the same problem recently with other packages; I'm quite confident this is a recent Vercel bug, perhaps related to https://github.com/vercel/vercel/pull/4814 ?

I also contacted support and they replied very quickly, but requested a minimal reproduction public repo, which I'm not interested in taking the time to put together.

**Comment by jaydenseric** [comment]
Closing because as far as I can tell there is nothing wrong with `apollo-upload-client`. Feel free to keep us all posted in comments though!

**Comment by djeusette** [comment]
Thanks for your help guys!  I'll keep you posted.

**Comment by jeroenvisser101** [comment]
@djeusette looking further in the PR Jayden mentioned, it seems this PR fixed something with a trailing slash: https://github.com/vercel/node-file-trace/pull/128

and from the logs you and I saw, this might be what breaks it.

**Comment by jaydenseric** [comment]
I filed a `@vercel/node-file-trace` bug here:

https://github.com/vercel/node-file-trace/issues/141#issue-666067917

**Comment by djeusette** [comment]
Hey guys, 

I built a minimal reproducible repository for them to check and I just got a reply:

```
Hello David,

This is solved in our canary branch. Follow these instructions:

    Use npm i -g vercel@canary​;
    Deploy with vercel​;
```

**Comment by djeusette** [comment]
That being said, it still fails...

**Comment by jaydenseric** [comment]
This should be fixed in `vercel@20.0.0-canary.5` 🎉
